core/agent_loop.py

```
@@ -51,135 +67,271 @@ def get_tool_schemas(self) -> List[Dict]:
+    
+    def _should_init_spec(self, task: str, routing_decision: RoutingDecision) -> bool:
+        """判断是否需要初始化 Spec"""
+        # 1. 检查是否为复杂任务
+        if routing_decision.task_type.value in ["code", "writing", "analyze"]:
+            # 2. 检查是否已存在 manifest
+            manifest = self.manifest_manager.load_manifest()
+            if manifest is None:
+                return True
+            
+            # 3. 检查任务类型是否匹配
+            # 简化：暂时总是返回 True，让用户决定
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)